internal/controller/isbservicerollout_controller.go

```diff
@@ -284,7 +293,7 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 		isbServiceRollout.Status.MarkDeployed(isbServiceRollout.Generation)
 	}
 
-	if common.DataLossPrevention {
+	if common.DataLossPrevention { // TODO: should this check be here? `&& upgradeStrategy == usde.UpgradeStrategyPPND`
```

Comment:
regarding the `TODO`, if we keep the function below named `processChildObjectWithoutDataLoss()` I think it can include Progressive. But we could also just change the name to something like `processChildObjectPPND()`

Reply:
That may be best so we can have a top level switch to choose between APPLY, PPND, or PROGRESSIVE.